tools/js-optimizer.js

```diff
@@ -4655,7 +4655,7 @@ function minifyLocals(ast) {
       if (asm) {
         return name in asmData.vars || name in asmData.params;
       } else {
-        return name in localNames;
+        return Object.prototype.hasOwnProperty.call(localNames, name);
```

Comment:
Can you just write `localNames.hasOwnProperty(name)`  ?

Reply:
Almost. The tricky thing is that there might be a function of that name, which means we'd have a property of that name on this "map", which would override the function we want to call.

I think we could fix that by using proper Maps everywhere, I'm not sure if there was a reason not to do that so far.